src/linters/rubo-cop.js

```diff
@@ -0,0 +1,70 @@
+const commandExists = require("../../vendor/command-exists");
+const { run } = require("../utils/action");
+
+/**
+ * https://rubocop.readthedocs.io
+ */
+class RuboCop {
+	static get name() {
+		return "RuboCop";
+	}
+
+	/**
+	 * Verifies that all required programs are installed. Exits the GitHub action if one of the
+	 * programs is missing
+	 *
+	 * @param {string} dir: Directory to run the linting program in
+	 */
+	static async verifySetup(dir) {
+		// Verify that Ruby is installed (required to execute RuboCop)
+		if (!(await commandExists("ruby"))) {
+			throw new Error("Ruby is not installed");
+		}
+
+		// Verify that RuboCop is installed
+		if (!(await commandExists("rubocop"))) {
+			throw new Error(`${this.name} is not installed`);
+		}
+	}
+
+	/**
+	 * Runs the linting program and returns the command output
+	 *
+	 * @param {string} dir: Directory to run the linting program in
+	 * @param {string[]} extensions: Array of file extensions which should be linted
+	 * @param {boolean} fix: Whether the linter should attempt to fix code style issues automatically
+	 * @returns {string}: Results of the linting process
+	 */
+	static lint(dir, extensions, fix = false) {
+		const output = JSON.parse(
+			run(`rubocop --format json ${fix ? "--auto-correct" : ""} ${dir}`, {
+				dir,
+				ignoreErrors: true,
+			}).stdout,
+		);
+
+		return JSON.stringify(
+			output.files.flatMap(f =>
+				f.offenses.map(o => ({
+					path: f.path,
+					firstLine: o.location.start_line,
+					lastLine: o.location.last_line,
+					message: o.message,
+				})),
+			),
+		);
```

Comment:
Could you please move the parsing logic into the `parseResults` function below? All the `lint` function is supposed to do is run the linting command and return the results as a string

Reply:
I originally had this in the other method but Rubocop add some version info to its json output which would make the generalized linter tests pretty fragile. We can move it back but we’d probably need specific rubocop tests instead of the param based ones.

Reply:
Ok, I see. Still, I think this is something that should be handled by the tests, not the source code. Maybe the `normalizeDates` function could be implemented in a more general way and take care of replacing the `metadata` part of the JSON output

Reply:
With that approach, the implementation of the `RuboCop` class can be very similar to the `ESLint` class, which also handles JSON output

Reply:
I'm gonna think about how I could improve the tests to avoid this. I'll get back to you :)

Reply:
Alright, I've improved and restructured the tests a bit. You can now specify either a `stdout` (for string equality) or `stdoutParts` (for substring checks) parameter. [This commit](https://github.com/samuelmeuli/lint-action/commit/574b45d3cee6f2bd756fe3b4cc8330a83191abfc) is probably a good example of how this could be done for RuboCop :)

Reply:
I pushed some commits before I had a chance to read this. I refactored the lint/parse methods and to fix the test issues I just allowed for a `formatResults` param to also be passed from the linter params which is then used in testing equality. Take a look and lmk what you think, it seems fair enough to me to let individual linters define their comparison instead of fixing it to str, array, ...